utils.go

```diff
@@ -57,3 +58,30 @@ func parseImageDestination(c *cli.Context, name string) (types.ImageDestination,
 	}
 	return nil, fmt.Errorf("Unrecognized image reference %s", name)
 }
+
+func parseDockerImage(img, certPath string, tlsVerify bool) (types.Image, error) {
+	s, err := newDockerImageSource(img, certPath, tlsVerify)
+	if err != nil {
+		return nil, err
+	}
+	return &dockerImage{src: s}, nil
+}
+
+// parseDockerImageName converts a string into a reference and tag value.
+func parseDockerImageName(img string) (reference.Named, string, error) {
```

Comment:
I did mean `docker_client.go`; `NewDockerImage{Source,Destination}` each calls both `parseDockerImageName` and `newDockerClient`. (We could even combine the two? Probably not, that seems uglier.)

The layering is `docker_image` → `docker_image_{source,destination}` → `docker_client`, putting it in `docker_image` causes a reference in the other direction.

Ultimately, if all of `docker_*` goes into the separate package, this does not matter that much of course :smile: